src/Java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/GregtechMetaTileEntity_SpargeTower.java

```diff
@@ -65,21 +63,21 @@
 				{"lll", "lll", "lll"}
 			}))
 			.addElement('b', ofChain(
-					ofHatchAdder(GregtechMetaTileEntity_SpargeTower::addEnergyInputToMachineList, getCasingIndex(), 1),
-					ofHatchAdder(GregtechMetaTileEntity_SpargeTower::addOutputToMachineList, getCasingIndex(), 1),
-					ofHatchAdder(GregtechMetaTileEntity_SpargeTower::addInputToMachineList, getCasingIndex(), 1),
-					ofHatchAdder(GregtechMetaTileEntity_SpargeTower::addMaintenanceToMachineList, getCasingIndex(), 1),
+					ofHatchAdder(GregtechMetaTileEntity_SpargeTower::addToMachineList, getCasingIndex(), 1),
```

Comment:
Duplicates?

Reply:
Whatcha mean?

Reply:
You are chaining 4 same structure checker.

Reply:
Why have four calls when you can just have one? AddToMachineList does the same thing.

Reply:
I probably don’t understand structurelib well enough tbh.

Reply:
Basically, all these elements passed to ofChain will be tried one by one until one returned true. This is functionally equivalent to `if (this.addToMachineList(getCasingIndex(), 1)))` to what you'd have in old style checkMachine. You can safely drop the other 3 duplicates without breaking anything.